space-plugins/nuxt-base/composables/useSettings.ts

```diff
@@ -0,0 +1,41 @@
+export const useSettings = async () => {
+	//TODO: pagination?
+
+	const { data, pending, error, refresh } = await useFetch('/api/settings');
+
+	const setSettings = async (body: Record<string, string>) => {
+		pending.value = true;
+
+		const mergedSettings = {
+			...data.value,
+			...body,
+		};
+
+		const {
+			data: changedData,
+			error: changedError,
+			pending: changedPending,
+		} = await useFetch('/api/settings', {
+			method: 'POST',
+			body: mergedSettings,
+		});
+
+		if (changedError.value) {
+			error.value = changedError.value;
+		}
+
+		if (changedData.value) {
```

Comment:
not sure if directly assigning the outer data from the useFetch is allowed in this style.

Reply:
I wouldn't touch `data` because it's meant to be returned and managed directly by `useFetch`. If updating the settings is successful here, I'd call `refresh()` to get the fresh data at line 4.